src/Compilers/Core/Portable/SourceGeneration/Nodes/SourceOutputNode.cs

```diff
@@ -19,13 +19,16 @@ internal sealed class SourceOutputNode<TInput> : IIncrementalGeneratorOutputNode
 
         private readonly Action<SourceProductionContext, TInput> _action;
 
-        public SourceOutputNode(IIncrementalGeneratorNode<TInput> source, Action<SourceProductionContext, TInput> action)
+        private readonly IncrementalGeneratorOutputKind _outputKind;
+
+        public SourceOutputNode(IIncrementalGeneratorNode<TInput> source, Action<SourceProductionContext, TInput> action, IncrementalGeneratorOutputKind outputKind)
```

Comment:
Consider an assert that `outputKind` is either `Source` or `Implementation`. Don't think that `PostInit` is valid here correct?